modules/objdetect/src/qrcode_encoder.cpp

```diff
@@ -0,0 +1,2102 @@
+    {
+        const int max_size = version_size;
+        const int version_block_width = 2;
+        const int xs_version[version_block_width][max_version_length] = {
+                { 5, 5, 5,
+                  4, 4, 4,
+                  3, 3, 3,
+                  2, 2, 2,
+                  1, 1, 1,
+                  0, 0, 0
+                },
+                { max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11
+                }
+        };
+        const int ys_version[version_block_width][max_version_length] = {
+                { max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11,
+                  max_size - 9, max_size - 10, max_size - 11
+                },
+                { 5, 5, 5,
+                  4, 4, 4,
+                  3, 3, 3,
+                  2, 2, 2,
+                  1, 1, 1,
+                  0, 0, 0,
+                }
+        };
+        for (int i = 0; i < version_block_width; i++)
+        {
+            for (int j = 0; j < max_version_length; j++)
+            {
+                if (version_reserved.at<uint8_t>(0, max_version_length - j - 1) == 0)
+                {
+                    masked.at<uint8_t>(ys_version[i][j], xs_version[i][j]) = 255;
+                }
+                else
+                {
+                    masked.at<uint8_t>(ys_version[i][j], xs_version[i][j]) = 0;
+                }
+            }
+        }
+    }
+}
+
+void QREncoder::structureFinalMessage()
+{
+    writeReservedArea();
+    writeData();
+    findAutoMaskType();
+    maskData(mask_type, masked_data);
+    formatGenerate(mask_type, format);
+    versionInfoGenerate(version_level, version_reserved);
+    fillReserved(format, masked_data);
+}
+
+void QREncoder::generatingProcess(Mat& final_result)
+{
+    vector<Mat> data_blocks, ecc_blocks;
+    if (!stringToBits())
+    {
+        return;
+    }
+    padBitStream();
+    eccGenerate(data_blocks, ecc_blocks);
+    rearrangeBlocks(data_blocks, ecc_blocks);
+    structureFinalMessage();
+    final_result = masked_data.clone();
+    const int border = 2;
+    copyMakeBorder(final_result, final_result, border, border, border, border, BORDER_CONSTANT, Scalar(255));
+}
+
+struct QRCodeEncoder::Impl
+{
+public:
+    Impl() {}
+    ~Impl() {}
+};
+QRCodeEncoder::QRCodeEncoder() : p(new Impl) {}
+QRCodeEncoder::~QRCodeEncoder() {}
+
+bool QRCodeEncoder::generate(cv::String input, cv::OutputArray output,
```

Comment:
`bool` result is not informative from the user perspective.

How to troubleshot the `false` result?

Reply:
> `cv::`

should be removed.